akka-testkit/src/main/scala/akka/testkit/javadsl/TestKit.scala

```diff
@@ -0,0 +1,427 @@
+   * given object. Wait time is bounded by the given duration, with an
+   * AssertionFailure being thrown in case of timeout.
+   */
+  def expectMsg[T](max: FiniteDuration, obj: T): T = tp.expectMsg(max, obj)
+
+  /**
+   * Receive one message from the test actor and assert that it equals the
+   * given object. Wait time is bounded by the given duration, with an
+   * AssertionFailure being thrown in case of timeout.
+   */
+  def expectMsg[T](max: FiniteDuration, obj: T, hint: String): T = tp.expectMsg(max, hint, obj)
+
+  /**
+   * Receive one message from the test actor and assert that the given
+   * partial function accepts it. Wait time is bounded by the given duration,
+   * with an AssertionFailure being thrown in case of timeout.
+   *
+   * Use this variant to implement more complicated or conditional
+   * processing.
+   */
+  def expectMsgPF[T](hint: String, f: JFunction[Any, T]): T = {
+    tp.expectMsgPF(hint = hint)(new JavaPartialFunction[Any, T] {
+      @throws(classOf[Exception])
+      override def apply(x: Any, isCheck: Boolean): T = f.apply(x)
+    })
+  }
+
+  /**
+   * Receive one message from the test actor and assert that the given
+   * partial function accepts it. Wait time is bounded by the given duration,
+   * with an AssertionFailure being thrown in case of timeout.
+   *
+   * Use this variant to implement more complicated or conditional
+   * processing.
+   */
+  def expectMsgPF[T](max: Duration, hint: String, f: JFunction[Any, T]): T = {
+    tp.expectMsgPF(max, hint)(new JavaPartialFunction[Any, T] {
+      @throws(classOf[Exception])
+      override def apply(x: Any, isCheck: Boolean): T = f.apply(x)
+    })
+  }
+
+  /**
+   * Same as `expectMsgClass(remainingOrDefault, c)`, but correctly treating the timeFactor.
+   */
+  def expectMsgClass[T](c: Class[T]): T = tp.expectMsgClass(c)
+
+  /**
+   * Receive one message from the test actor and assert that it conforms to
+   * the given class. Wait time is bounded by the given duration, with an
+   * AssertionFailure being thrown in case of timeout.
+   */
+  def expectMsgClass[T](max: FiniteDuration, c: Class[T]): T = tp.expectMsgClass(max, c)
+
+  /**
+   * Same as `expectMsgAnyOf(remainingOrDefault, obj...)`, but correctly treating the timeFactor.
+   */
+  def expectMsgAnyOf[T](objs: JIterable[T]): T = tp.expectMsgAnyOf(Util.immutableSeq(objs): _*)
+
+  /**
+   * Receive one message from the test actor and assert that it equals one of
+   * the given objects. Wait time is bounded by the given duration, with an
+   * AssertionFailure being thrown in case of timeout.
+   */
+  def expectMsgAnyOf[T](max: FiniteDuration, objs: JIterable[T]): T = tp.expectMsgAnyOf(max, Util.immutableSeq(objs): _*)
+
+  /**
+   * Same as `expectMsgAllOf(remainingOrDefault, obj...)`, but correctly treating the timeFactor.
+   */
+  def expectMsgAllOf[T](objs: JIterable[T]): Array[T] = tp.expectMsgAllOf(Util.immutableSeq(objs): _*).toArray(classTag[Any]).asInstanceOf[Array[T]]
```

Comment:
I kept the returning array since it was used in some unit tests. How about removing the type parameter `T` here:

```scala
def expectMsgAllOf(objs: JIterable[Any]): Array[Any] = tp.expectMsgAllOf(Util.immutableSeq(objs): _*).toArray(classTag[Any])
```